### ISSUE TITLE
Combobox blank after selecting the same option twice

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-components#4738]
### Actual Behavior

When a combobox option is selected in single select mode, selecting it again does not show the option as selected. It stays blank with the text input visible. 

### Expected Behavior

Selecting an option should show that option as selected. This will keep the combobox consistent with calcite select and consistent internally with itself as after selecting a new option.

### Reproduction Sample

https://codepen.io/paulcp/pen/abqXgqE

### Reproduction Steps

1. Open the combobox
2. Select Altitude (already selected)

### Reproduction Version

1.0.0-beta.83

### Relevant Info

Looking at different versions, it seems like this has behaved very differently in different versions. 

- beta.83 the combobox will close, the input remains focused and active, and shows as blank
- beta.81 will close the options, hide the input, but show as blank
- beta.69 and earlier worked as expected

### Regression?

1.0.0-beta.69

**Comment by macandcheese** [comment]
@paulcpederson would one of the suggestions for an additional "mode" or type of Combobox selection work here? Both UX could be desirable I think... https://github.com/Esri/calcite-components/issues/3074

**Comment by paulcpederson** [comment]
This issue is exclusively tied to single select to me. In multi select I think the behavior is fine, but if I'm using single select and I select an item, I expect it to behave like a select menu. Selecting one item vs another should not have different behavior, IMO.

**Comment by macandcheese** [comment]
We could look at the behavior of MUI as an example - they by default have a clear affordance that is opt-out with `disableClear` - this could be combined with fixing the regression to solve all the requests in - https://github.com/Esri/calcite-components/issues/3074

**Comment by geospatialem** [comment]
For the dev taking this on - the bug is still present in `1.2.0`, and when a value is already selected, should remain in the `combobox`, even if its selected again. Showing current behavior, and expected (from [MUI](https://mui.com/material-ui/react-autocomplete/#combo-box)).

It also looks like the `value` prop isn't working, perhaps related to a fix?

Updated Codepen: https://codepen.io/geospatialem/pen/QWZLPyQ

## Current behavior (still in `1.2.0`)
![repro](https://user-images.githubusercontent.com/5023024/230116979-86fe7748-4d44-4017-b324-15dbce291de5.gif)

## Expected behavior (from [MUI](https://mui.com/material-ui/react-autocomplete/#combo-box))
![repro-expected](https://user-images.githubusercontent.com/5023024/230117728-3479483c-99f4-440f-8a9f-a4941b149465.gif)

**Comment by macandcheese** [comment]
Need to add the clear affordance - if above is implemented without adding the clear affordance, users won't be able to deselect an item.

**Comment by ashetland** [comment]
Figma file available as part of related issue #3074.

**Comment by geospatialem** [comment]
Verified in `1.4.0-next.22`

![combobox-same-option-blank](https://github.com/Esri/calcite-components/assets/5023024/eadf7c33-4e31-4e3a-9412-634abe1bd857)

**Comment by jcfranco** [comment]
This ended up breaking existing user workflows, so we'll be reverting and addressing at a later release. Hopefully, in a non-breaking manner. 🤞 cc @ashetland @SkyeSeitz @macandcheese

**Comment by jcfranco** [comment]
Sharing a workaround for now: https://codepen.io/jcfranco/pen/JjmVGmo?editors=1001

**Comment by geospatialem** [comment]
Research is needed to determine a good approach which should include impacting existing workflows per [Franco's note above](https://github.com/Esri/calcite-components/issues/4738#issuecomment-1564793661).

**Comment by jcfranco** [comment]
> Add a disableClear property to allow users to opt-out of displaying that clear "x" affordance, to basically create a "single-persist" mode without needing to add another mode option

@macandcheese @ashetland In order to avoid the breaking change, we would need to add `single-persist` as a mode. We would have to figure out how `disableClear` affects this experience. I have a draft PR for this (linked above) if you'd like to test drive this.

**Comment by macandcheese** [comment]
If the functionality of `single-persist` removes the need for `disableClear`, maybe we can deprecate the latter? Using the `selectionMode` to determine functionality matches other components.

**Comment by ashetland** [comment]
Agree. I'm not sure we need `disableClear` if there's a `single-persist` mode.

**Comment by jcfranco** [comment]
Would the goal be to have the current behavior of `disableClear` kick in automatically based on the selection mode then?

**Comment by jcfranco** [comment]
For posterity, we'll proceed to add `single-persist` mode and `disableClear` will keep on truckin' on. 🛻

**Comment by geospatialem** [comment]
The mouse event is verified with `1.7.0-next.9`, however keyboard user functionality behaves a bit differently. 

@jcfranco Would we expect the same behavior for keyboard users? Here is a [codepen for testing](https://codepen.io/geospatialem/pen/vYvLYEO).

**Comment by jcfranco** [comment]
@geospatialem Forgot to focus on keyboard toggling. It should be good to go now! 🚀

**Comment by geospatialem** [comment]
Verified with both mouse and keyboard selections in `1.7.0-next.16`.
